scripts/globals/dynamis.lua

```diff
@@ -292,37 +305,46 @@ dynamis.entryNpcOnTrigger = function(player, npc)
+    elseif player:hasKeyItem(xi.ki.PRISMATIC_HOURGLASS) and player:getMainLvl() < xi.settings.DYNA_LEVEL_MIN then
+        player:messageSpecial(ID.text.PLAYERS_HAVE_NOT_REACHED_LEVEL)
+    -- default message always prints except in cases above and not for shrouded sand or winning cs
+    elseif mask ~= 1 and player:getCharVar(info.beatVar) ~= 1 then
+        player:messageSpecial(ID.text.DYNA_NPC_DEFAULT_MESSAGE)
+    end
 
-    -- victory cutscene
-    elseif player:getCharVar(info.winVar) == 1 then
-        player:startEvent(info.csWin, info.winKI)
-
-    -- dynamis entry
-    elseif not info.reqs or info.reqs(player) then
-        local realDay = os.time()
-        local dynaWaitxDay = player:getCharVar("dynaWaitxDay")
-
-        if player:getMainLvl() < xi.settings.DYNA_LEVEL_MIN then
-            player:messageSpecial(ID.text.PLAYERS_HAVE_NOT_REACHED_LEVEL, xi.settings.DYNA_LEVEL_MIN)
-        elseif (dynaWaitxDay + xi.settings.BETWEEN_2DYNA_WAIT_TIME * 60 * 60) < realDay then
-            -- params: bit, cutscene option, Prismatic Hourglass KI, sJob option, junk, Shrouded Sand KI, Timeless Hourglass item ID, Perpetual Hourglass item ID
-            player:startEvent(info.csDyna, info.csBit, arg3(player, info.csBit), xi.ki.PRISMATIC_HOURGLASS, 1, 0, xi.ki.VIAL_OF_SHROUDED_SAND, 4236, 4237)
-        else
-            local dayRemaining = math.floor(((dynaWaitxDay + xi.settings.BETWEEN_2DYNA_WAIT_TIME * 60 * 60) - realDay) / 3456)
-            player:messageSpecial(ID.text.YOU_CANNOT_ENTER_DYNAMIS, dayRemaining, info.csBit)
-        end
+    -- all cutscenes and menus are blocked behind base requirements; 'or mask == 1' needs to be set to access shroud cs after zoning into xarcabard
+    if not tavnaziaFirst and player:getMainLvl() >= xi.settings.DYNA_LEVEL_MIN and (player:hasKeyItem(xi.ki.PRISMATIC_HOURGLASS) or mask == 1) then
 
-    -- standard dialog
-    else
-        player:messageSpecial(ID.text.DYNA_NPC_DEFAULT_MESSAGE)
+        -- shrouded sand cutscene
+        if mask == 1 and info.csVial and not player:hasKeyItem(xi.ki.VIAL_OF_SHROUDED_SAND) then
+            player:startEvent(info.csVial)
+
+        -- victory cutscene
+        elseif player:getCharVar(info.beatVar) == 1 then
+            player:startEvent(info.csBeat, info.beatKI)
+
+        -- dynamis entry
+        elseif not info.reqs or info.reqs(player) then
+            local realDay = os.time()
+            local dynaWaitxDay = player:getCharVar("dynaWaitxDay")
+            local sjobOption = info.csBit > 6 and 1 or 0
+
+            if (dynaWaitxDay + xi.settings.BETWEEN_2DYNA_WAIT_TIME * 60 * 60) < realDay then
+                -- params: bit, cutscene option, Prismatic Hourglass KI, sJob option, junk, Shrouded Sand KI, Timeless Hourglass item ID, Perpetual Hourglass item ID
+                player:startEvent(info.csMenu, info.csBit, arg3(player, info.csBit), xi.ki.PRISMATIC_HOURGLASS, sjobOption, 0, xi.ki.VIAL_OF_SHROUDED_SAND, 4236, 4237)
```

Comment:
The only real change is here is the addition of `sjobOption` to provide proper menu selections for cities and dreamlands alike. The rest of green comes from moving the default dialogue up.